src/test_utils.rs

```diff
@@ -1351,6 +1351,24 @@ async fn write_msg(context: &Context, prefix: &str, msg: &Message, buf: &mut Str
     .unwrap();
 }
 
+/// When dropped after a test failure,
+/// prints a note about a possible false-possible caused by SystemTime::shift().
+pub(crate) struct TimeShiftFalsePositiveNote;
+impl Drop for TimeShiftFalsePositiveNote {
+    fn drop(&mut self) {
+        if std::thread::panicking() {
```

Comment:
It may not panic at this point but return `Error` in some tests also, but idk how to catch that. Anyway, the proposed approach is better than the previous `assert!` in the only test